### PR TITLE
334 improve chloropleth legend

### PR DESCRIPTION
**Pull request by ptgott**
Addresses #334

### CODE REVIEW

File: docs/tool/js/core/housing-insights.js [hunk 1]
```diff
@@ -296,6 +296,11 @@ var setSubs = controller.controlSubs.setSubs,
 
  //TODO should these be wrapped up into a 'helpers' namespace? Or maybe the new utils section?
 
+// HELPER roundTo
```
Comment (NealHumphrey) [inline docs/tool/js/core/housing-insights.js]: For this and the comma-separated thing, I realized that D3 already provides this functionality - we should use the d3.format() function to do this work for us. https://github.com/d3/d3-format

File: docs/tool/js/views/chloropleth-legend.js [hunk 2]
```diff
@@ -36,11 +36,9 @@ var chloroplethLegend = {
           return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ","); //adds thousands separators, from https://stackoverflow.com/questions/2901102/how-to-print-a-number-with-commas-as-thousands-separators-in-javascript
```
Comment (NealHumphrey) [inline docs/tool/js/views/chloropleth-legend.js]: Again, using d3.format() would be cleaner - not blocking this PR unless you want to change it now, but noted for future refactoring.

File: docs/tool/js/views/map-view.js [hunk 3]
```diff
@@ -135,9 +135,23 @@
...
+                    default:
+                        return Math.round(val);
+                }
+            }
+
             this.stops = new Array(CHLOROPLETH_STOP_COUNT).fill(" ").map(function(el, i){
                 var stopIncrement = MAX_DOMAIN_VALUE/(CHLOROPLETH_STOP_COUNT - 1);
-                var domainValue = Math.round((MAX_DOMAIN_VALUE - (stopIncrement * i))*1000)/1000; //stupid javascript rounding. Close enough for this purpose. 
+                var domainValue = MAX_DOMAIN_VALUE - (stopIncrement * i);
+                var domainValue = roundedVersionOf(domainValue);
```
Comment (NealHumphrey) [inline docs/tool/js/views/map-view.js]: do we want the second `var` here? Does it even make a difference to re-var the variable?

**Comment by ptgott** [comment]
Thanks for your review @NealHumphrey! I'll merge this now and leave an issue for later.

**Pull request merged**